src/Illuminate/Support/Str.php

```diff
@@ -531,6 +531,17 @@ public static function uuid()
         return Uuid::uuid4();
     }
 
+    /**
+     * Return true if the param is a string, or an object that maybe treated as string.
+     * @param $value
+     * @return bool
+     */
+    public static function isStringable($value)
+    {
+        return is_string($value) ||
+            (is_object($value) && is_callable([$value, '__toString']));
```

Comment:
> If the target class has __call() magic function implemented, then is_callable will ALWAYS return TRUE for whatever method you call it.

You should use `method_exists` instead

Reply:
Also, I believe the way to catch all the possible stringable values is:

`return $value === null || is_scalar($value) || method_exists($value, '__toString');`